GoogleForCreators/web-stories-wp
# Pinch to Zoom (desktop)

## Summary
Adds pinch to zoom feature for desktop.


## Relevant Technical Choices
Due to performance & simplicity reasons, currently allows pinching to change the zoom by 25% only.

Ideally, we'd use `transform: scale` on the `fullbleedContainer` and only update the state every X time, however, that’s a bit complex due to all kind of related parts that are influenced and dynamically displayed/changed depending on the zoom level. For example, scrollbars, page navigation, the scroll offset, etc. making it “jump” when it’s re-rendered “correctly”. Additionally, the scaled page would flow over the page thumbnails, etc. It’s probably doable nicely somehow, however, it would need more investigation and would be a larger feature and was confirmed by the Product team as a nice-to-have only.


## To-do



## User-facing changes
![step-by-step](https://user-images.githubusercontent.com/3294597/118941757-589be280-b952-11eb-83a4-a74224d2259d.gif)



## Testing Instructions
1. Move the cursor above the Page and start pinching.
2. Verify you can change the zoom level by 25% by pinching, max value being 300%, min value 25%.


### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Checklist




---



Fixes #6179
